code/game/turfs/flooring/flooring.dm

```diff
@@ -35,52 +39,217 @@ var/list/flooring_types
 	var/build_time = 0  // BYOND ticks.
 
 	var/descriptor = "tiles"
-	var/flags
+	var/flags = TURF_CAN_BURN | TURF_CAN_BREAK
 	var/can_paint
 
+	var/is_plating = FALSE
+
+	//Plating types, can be overridden
+	var/plating_type = /decl/flooring/reinforced/plating
```

Comment:
Plating type determines what a floor becomes when broken. This is used extensively to control how the plating types break down into other platings